src/bp/ui-admin/src/telemetry.ts

```diff
@@ -239,28 +223,13 @@ export function getTelemetryPackage(event_type: string, data: dataType, name: st
     bp_release: info.bp_release,
     bp_license: info.bp_license,
     event_type: event_type,
+    source: 'client',
     event_data: getDataCluster(data)
   }
 }
 
-function sendTelemetry(data: TelemetryPackage, event: string) {
-  axios
-    .post(endpointMock, data, {
-      headers: {
-        'content-type': 'application/json',
-        withCredentials: false,
-        timeout: 1000,
-        transformRequest: [
-          (data, headers) => {
-            return JSON.stringify(data)
-          }
-        ]
-      }
-    })
-    .then(res => {
-      addTimeout(event, ms('8h'))
-    })
-    .catch(err => {
-      changeLock(event)
-    })
+async function sendTelemetry(data: TelemetryPackage, event: string) {
+  const res = await axios.post(endpoint, data, corsConfig)
+  console.log('info sent', res, data)
+  addTimeout(event, ms('8h'))
```

Comment:
what does that function do ? isn't it already handled by set interval ?

Reply:
It records the moment when the package can be sent again and puts locks for the same amount of time. If the browser is closed then it keeps a record of the moment when it can be sent again.

Reply:
setInterval is not used in this part of the code. setInterval is used in the function sendServerPackageLoop, because it is better for the use case of the middleware than the actual telemetry of the client side.

Reply:
I used setTimeout because it is easier to implement with store.subscribe than to create an implementation with setInterval